sql/opt/exec/explain: TestMaximumMemoryUsage failed

sql/opt/exec/explain.TestMaximumMemoryUsage [failed](https://tanzanite.cluster.engflow.com/invocations/default/92a789ea-5480-4e3f-8024-1e66e0fba98e?testReportRun=4&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL29wdC9leGVjL2V4cGxhaW46ZXhwbGFpbl90ZXN0) on release-24.3.12-rc @ [539424eb7082f9fc04d86a60bb961c9f74596be1](https://github.com/cockroachdb/cockroach/commits/539424eb7082f9fc04d86a60bb961c9f74596be1):

Fatal error:

```
panic: test timed out after 1m0s
running tests:
	TestMaximumMemoryUsage (48s)

```
Stack: 

```
goroutine 94926 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestMaximumMemoryUsage
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestMaximumMemoryUsage1281698056
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMaximumMemoryUsage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-49703

--- a/pkg/sql/opt/exec/explain/BUILD.bazel
+++ b/pkg/sql/opt/exec/explain/BUILD.bazel
@@ -52,7 +52,7 @@ go_library(
 
 go_test(
     name = "explain_test",
-    size = "small",
+    size = "medium",
     srcs = [
         "explain_factory_test.go",
         "main_test.go",

